src/Common/Extensions/StringExtensions.cs

```diff
@@ -24,7 +24,10 @@ public static bool EqualsIgnoreCase(this string a, string b)
             return string.Equals(a, b, StringComparison.OrdinalIgnoreCase);
         }
 
-        public static string JoinKeyValuePairs(IDictionary<string, string> pairs, char pairSep = ' ', char kvSep = '=')
+        public static string JoinKeyValuePairs(
+            IDictionary<string, string> pairs,
+            string pairSep = " ",
```

Comment:
The change is from using `' '` to `"  "`...In this PR I made a change to Detector project where it now adds couple of files `FileExtensions.cs` and `StringExtensions.cs` as 'links' from the `Common` project. This way we need not duplicate code.
However there are some caveats with this approach. For example, since the files here get build either in 'netstandard2.0' or 'netcoreapp3.1' projects, we need to have code which only works in both projects. Here using `" "` works because it is supported in 'netstandard2.0' and since 'netcoreapp3.1' supports 'netstandard2.0', things just work.

Summary is to stick to API which is supported in 'netstandard2.0' to avoid errors

Reply:
Scratch the above comment since I noticed a problem in doing so..Since Detector is being referenced in BuildScriptGenerator too, there was a conflict since the same type name existed in both the assemblies. So I instead split the projects into multiple ones.

Reply:
Just to summarize the new changes here to make sure, so now we have split into `Oryx.Common` and `BuildScriptGenerator.Common`. `BuildScriptGenerator.Common` and `Detector ` are referencing `Oryx.Common`. `BuildScriptGenerator `and `Test.Common` are referencing `BuildScriptGenerator.Common`.

Reply:
BTW, in the VS, it doesn't show as` Oryx.Common` but as `Common`, and in the folder I can see it..so maybe need to configure csproj

Reply:
Good catch. I assigned AssemblyName to Microsoft.Oryx.Common now. Thanks.